bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/AbstractExpressionPart.java

```diff
@@ -0,0 +1,180 @@
+     */
+    public void setValueSet(BoundedIntegerSet set) {
+        this.valueSet = set;
+    }
+
+    abstract BoundedIntegerSet initializeValueSet();
+
+    /**
+     * A <code>BoundedIntegerSet</code> is a set of Integers that is bound by a minimum and maximum value, that does
+     * allow or not allow negaive values, and that is either starts from index 0 or 1
+     *
+     * @author Karel Goderis
+     *
+     */
+    protected class BoundedIntegerSet extends TreeSet<Integer> {
+        private static final long serialVersionUID = 19296179649170335L;
+        protected final int max;
+        protected final int min;
+        protected final boolean negative;
+        protected boolean is1indexed;
+
+        BoundedIntegerSet(final int min, final int max, final boolean negativeValuesAllowed, final boolean is1indexed) {
+            this.min = min;
+            this.max = max;
+            this.negative = negativeValuesAllowed;
+            this.is1indexed = is1indexed;
+        }
+
+        /**
+         * Validates if an integer can be added to the set
+         *
+         * @param integer - the integer to validate
+         */
+        private void validate(final Integer integer) {
+            if (negative) {
+                final int abs = Math.abs(integer);
+                if (abs < min || abs > max) {
+                    throw new IllegalArgumentException("Invalid integer value (value not in range [" + (-max) + ", "
+                            + -min + "] U [" + min + ", " + max + "])");
+                }
+            } else {
+                if (integer < min || integer > max) {
+                    throw new IllegalArgumentException(
+                            "Invalid integer value (value not in range [" + min + ", " + max + "])");
+                }
+            }
+        }
+
+        @Override
+        public boolean add(final Integer e) {
+            validate(e);
+            return super.add(e);
+        }
+
+        /**
+         * Adds a series of integers to the set, starting from start, and ending at end, with a given increment
+         *
+         * @param start - value to start at
+         * @param end - value to end at (inclusive)
+         * @param increment - increment to advance
+         * @return true if the addition of integers was successful
+         */
+        public boolean add(final Integer start, final Integer end, final Integer increment) {
+
+            if (end != null && end > max) {
+                throw new IllegalArgumentException(
+                        "Invalid integer value (end not in range [" + min + ", " + max + "])");
+            }
+
+            int stopAt;
+            if (end == null) {
+                stopAt = this.max;
+            } else {
+                stopAt = end;
+            }
+
+            int startAt;
+            if (start == null) {
+                startAt = this.min;
+            } else {
+                startAt = start;
```

Comment:
You test end > max above but do not check for (start != null && start < min). Is this be done by intention?